HHS/Head-Start-TTADP
# [TTAHUB-2238] Filter for recipients without activity in range

## Description of change

This adds a new filter to the RTR search called 'Recipients without TTA'. 

Activity is determined by using the start and end date of the linked AR's.

![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/e649af36-261e-49a4-83d7-ef5e8b383c8e)

It has the following options: 
- Last thirty days
- Last three months
- Last six months
- Year to date
- Custom date range which allows users to enter a start and end date. 

**Note:** There is a linked mockup that shows a completely different layout of the date filter. If we need this change also it should be its own ticket as its not in this JIRA's requirements.

## How to test

- Review the code
- All tests pass
- Test the new filter from the RTR search
- Make sure the existing date filters worked as they did before.

Something like the below query can be used to find activity in the range (the opposite).
`  SELECT
	  	ar.id,
		ar."startDate",
		ar."endDate"
	  FROM "ActivityReports" ar WHERE id IN (
	  SELECT "activityReportId" 
	  FROM "ActivityReportGoals"
	  WHERE "goalId" IN (
	  SELECT id from "Goals" WHERE "grantId" = (
		 SELECT id FROM "Grants" WHERE "recipientId" IN (
		  SELECT id FROM "Recipients" WHERE "name" = 'City of New Haven, Inc.'
	  ) AND status = 'Active')))
	  	 AND (ar."startDate", ar."endDate") OVERLAPS ('01/01/2022','11/28/2023')
	  order by ar."startDate", ar."endDate"`

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2338


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
